.github/workflows/build.yml

```diff
@@ -49,18 +49,18 @@ jobs:
 
       - uses: Swatinem/rust-cache@v2
         with:
-          key: cargo-${{ inputs.python-architecture }}-${{ inputs.os }}-${{ inputs.msrv }}
+          key: cargo-${{ inputs.python-architecture }}
```

Comment:
Apparently the cache automatically includes host and rustc release, so I removed these from the key.

https://github.com/Swatinem/rust-cache/tree/v2/#cache-details